### DISCUSSION TITLE
Expand Search Results Group Based on Search Query Results

### DISCUSSION THREAD

**Discussion by dontthroworanges** [microsoft-search/pnp-modern-search#4166] [Q&A]
Is it possible to expand a group in the Search Results webpart to expose the result of a Search Box query? While the groups are filtered based on the query, the user still needs to click down through the various levels to expose the actual result. Any assistance would be greatly appreciated. Thanks!

**Comment by kasperbolarsen** [comment]
Can you expand on this requirement, I can't follow you.
If I search for "Contoso" and the Results web part shows documents related to Contoso, grouped by "approval state" or "created by" or similar, how should we be able to figure out which group to open?

**Reply by dontthroworanges** [reply]
Hi Kasper, 
For your example I would say that both groups should be expanded. The Results web part is already filtering to only the specific groups where that content is found, but instead of looking like this: 

![image](https://github.com/user-attachments/assets/96801158-1b8a-4752-9cb5-ccf794c22dc7)

It should look like this without the user having to click to expand each category that the result is listed under:

![image](https://github.com/user-attachments/assets/92147bc3-4a78-4cab-9fe8-1230696353ea)

Does that clarify things a bit more? Let me know if you require additional info. Thanks!

**Reply by kasperbolarsen** [reply]
I do kind of understand the request, and can see that it would make sense when the number of groups is low. However, if we have 50+ groups and they all are expanded, then the part of the purpose of groups centered around fast visibility is lost IMHO.

**Reply by dontthroworanges** [reply]
Hi @kasperbolarsen, 
I totally agree that the more levels you have, the less useful it becomes. However, for this use case there will only be a maximum of three levels. So, with that in mind, is there a way to customize my results template to make this possible? 

Thanks!
